src/tests/unit_tls.cpp

```diff
@@ -218,164 +223,216 @@ Test::Result test_tls_handshake(Botan::TLS::Protocol_Version offer_version,
+               s2c_traffic.clear();
+               server_recv.clear();
+               client_recv.clear();
+               client_sent.clear();
+               server_sent.clear();
+
+               // TLS::Server object constructed by legacy constructor.
+               server.reset( 
+                  new Botan::TLS::Server(queue_inserter(s2c_traffic),
+                                         queue_inserter(server_recv),
+                                         alert_cb_with_data, 
+                                         handshake_complete,
+                                         server_sessions,
+                                         creds,
+                                         policy,
+                                         rng,
+                                         next_protocol_chooser,
+                                         false));
+
+               // TLS::Client object constructed by legacy constructor.
+               client.reset( 
+                  new Botan::TLS::Client(queue_inserter(c2s_traffic),
+                                         queue_inserter(client_recv),
+                                         alert_cb_with_data,
+                                         handshake_complete,
+                                         client_sessions,
+                                         creds,
+                                         policy,
+                                         rng,
+                                         Botan::TLS::Server_Information("server.example.com"),
+                                         offer_version,
+                                         protocols_offered));
                }
 
-            if(handshake_done && (client.is_closed() || server.is_closed()))
-               break;
-
-            if(client.is_active() && client_sent.empty())
+            while(true)
                {
-               // Choose a len between 1 and 511
-               const size_t c_len = 1 + rng.next_byte() + rng.next_byte();
-               client_sent = unlock(rng.random_vec(c_len));
+               ++rounds;
 
-               // TODO send in several records
-               client.send(client_sent);
-               }
+               if(rounds > 25)
+                  {
+                  if(r <= 2)
+                     result.test_failure("Still here after many rounds, deadlock?");
+                  break;
+                  }
 
-            if(server.is_active() && server_sent.empty())
-               {
-               result.test_eq("server protocol", server.next_protocol(), "test/3");
+               if(handshake_done && (client->is_closed() || server->is_closed()))
+                  break;
 
-               const size_t s_len = 1 + rng.next_byte() + rng.next_byte();
-               server_sent = unlock(rng.random_vec(s_len));
-               server.send(server_sent);
-               }
+               if(client->is_active() && client_sent.empty())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if|for|do|while|switch statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AS134)